src/main.cpp

```diff
@@ -2540,16 +2631,35 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
+            if (!viewTempCache.HaveInputs(tx)) {
+                // ** In the case of IBD we have to check if the height has changed while we're checking inputs because this thread can continue for a
+                // short while after it has been interrupted and meanwhile the view of the UTXO could have changed after another thread has updated the tip.
+                // ** Additionally and more importantly in the event of a big block DDOS attack, by checking the chain height here, we ensure that
+                //    the thread that is validating the big block will immediately exit and finish up (while still keeping the block on disk in the
+                //    case of a reorg) as soon as the first block makes it through and wins the validation race.
```

Comment:
@dgenr8 Yes you are correct in that it needs to exit, but still the code has to be here anyway.  The reason is again, we can not allow the "return DOS 100" to happen if this is a thread that has merely lost a validation race or a reorg is underway, and if it has lost the race, or reorg is happening, then the next call to viewTempCache.haveInputs(tx) will fail and we will execute this section of code anyway. 

And of course if somehow the thread is done checking inputs then it will be terminated further down stream.